website/static/js/registerNode.js

```diff
@@ -6,28 +6,34 @@ var $ = require('jquery');
 var bootbox = require('bootbox');
 
 var preRegisterMessage =  function(title, parentTitle, parentUrl) {
-    return 'You are about to Register the component "' + title +
-        '" and everything that is inside it. This will not register' +
-        ' your larger project "' + parentTitle + '" and its other components.' +
-        ' If you want to register the parent project, please go <a href="' +
-        parentUrl + '">here.</a>';
+    if (parentUrl) {
+        return 'You are about to Register the component "' + title +
+            '" and everything that is inside it. This will not register' +
+            ' your larger project "' + parentTitle + '" and its other components.' +
+            ' If you want to register the parent project, please go <a href="' +
+            parentUrl + '">here.</a>';
+    } else {
+        return 'You are about to Register the project "' + title +
+            '" and everything that is inside it. If you would prefer to register ' +
+            'just a particular component of "' + title + '", please click back ' +
+            'and navigate to that component and then initiate registration.';
+    }
 };
 
 $(document).ready(function() {
     $('#registerNode').click(function(event) {
         var node = window.contextVars.node;
         var target = event.currentTarget.href;
-        if (node.parentExists) {
-            event.preventDefault();
-            var title = node.title;
-            var parentTitle = node.parentTitle;
-            var parentRegisterUrl = node.parentRegisterUrl;
-            
-            bootbox.confirm(preRegisterMessage(title, parentTitle, parentRegisterUrl), function(result) {
-                if (result) {
-                    window.location.href = target;
-                }
-            });
-        }        
+
+        event.preventDefault();
+        var title = node.title;
+        var parentTitle = node.parentTitle;
+        var parentRegisterUrl = node.parentRegisterUrl;
+
+        bootbox.confirm(preRegisterMessage(title, parentTitle, parentRegisterUrl), function (result) {
```

Comment:
See https://cosdev.readthedocs.org/en/latest/osf/guidelines.html#dialogs for correct bootbox usage.